tests/cache_/tests.py

```diff
@@ -0,0 +1,969 @@
+
+        self.assertEqual(cache.incr_version("answer", version=2), 3)
+        self.assertIsNone(cache.get("answer"))
+        self.assertIsNone(cache.get("answer", version=1))
+        self.assertIsNone(cache.get("answer", version=2))
+        self.assertEqual(cache.get("answer", version=3), 42)
+
+        caches["v2"].set("answer2", 42)
+        self.assertEqual(caches["v2"].get("answer2"), 42)
+        self.assertIsNone(caches["v2"].get("answer2", version=1))
+        self.assertEqual(caches["v2"].get("answer2", version=2), 42)
+        self.assertIsNone(caches["v2"].get("answer2", version=3))
+
+        self.assertEqual(caches["v2"].incr_version("answer2"), 3)
+        self.assertIsNone(caches["v2"].get("answer2"))
+        self.assertIsNone(caches["v2"].get("answer2", version=1))
+        self.assertIsNone(caches["v2"].get("answer2", version=2))
+        self.assertEqual(caches["v2"].get("answer2", version=3), 42)
+
+        with self.assertRaises(ValueError):
+            cache.incr_version("does_not_exist")
+
+        cache.set("null", None)
+        self.assertEqual(cache.incr_version("null"), 2)
+
+    def test_decr_version(self):
+        cache.set("answer", 42, version=2)
+        self.assertIsNone(cache.get("answer"))
+        self.assertIsNone(cache.get("answer", version=1))
+        self.assertEqual(cache.get("answer", version=2), 42)
+
+        self.assertEqual(cache.decr_version("answer", version=2), 1)
+        self.assertEqual(cache.get("answer"), 42)
+        self.assertEqual(cache.get("answer", version=1), 42)
+        self.assertIsNone(cache.get("answer", version=2))
+
+        caches["v2"].set("answer2", 42)
+        self.assertEqual(caches["v2"].get("answer2"), 42)
+        self.assertIsNone(caches["v2"].get("answer2", version=1))
+        self.assertEqual(caches["v2"].get("answer2", version=2), 42)
+
+        self.assertEqual(caches["v2"].decr_version("answer2"), 1)
+        self.assertIsNone(caches["v2"].get("answer2"))
+        self.assertEqual(caches["v2"].get("answer2", version=1), 42)
+        self.assertIsNone(caches["v2"].get("answer2", version=2))
+
+        with self.assertRaises(ValueError):
+            cache.decr_version("does_not_exist", version=2)
+
+        cache.set("null", None, version=2)
+        self.assertEqual(cache.decr_version("null", version=2), 1)
+
+    def test_custom_key_func(self):
+        # Two caches with different key functions aren't visible to each other
+        cache.set("answer1", 42)
+        self.assertEqual(cache.get("answer1"), 42)
+        self.assertIsNone(caches["custom_key"].get("answer1"))
+        self.assertIsNone(caches["custom_key2"].get("answer1"))
+
+        caches["custom_key"].set("answer2", 42)
+        self.assertIsNone(cache.get("answer2"))
+        self.assertEqual(caches["custom_key"].get("answer2"), 42)
+        self.assertEqual(caches["custom_key2"].get("answer2"), 42)
+
+    @override_settings(
+        CACHE_MIDDLEWARE_ALIAS=DEFAULT_CACHE_ALIAS,
```

Comment:
Use [@modify_settings](https://docs.djangoproject.com/en/dev/topics/testing/tools/#django.test.SimpleTestCase.modify_settings) for appending to `INSTALLED_APPS`.